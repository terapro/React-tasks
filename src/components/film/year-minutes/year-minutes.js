import React from 'react';
import PropTypes from 'prop-types';
import {Label} from 'src/components/common/label/label.js';

export const YearMinutes = ({year, minutes}) => (
    <div className='year-min'>
      <Label content={year.slice(0,4)} labelWhite labelBold />
      <Label content={minutes + ' min'} labelWhite labelBold/>
    </div>
  );

YearMinutes.propTypes = {
  year: PropTypes.string,
  minutes: PropTypes.number
};

YearMinutes.defaultProps = {
  year: 'unknown date',
  minutes: '0'
};