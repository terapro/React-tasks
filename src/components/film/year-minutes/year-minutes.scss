.year-min {
  display: flex;
}

.year-min > div {
  margin: 0 $elementPadding*2 $elementPadding $elementPadding*2
}

.year-min > div:first-child {
  margin: 0 0 $elementPadding 0
}

.year-min > div:last-child {
  margin: 0 0 $elementPadding $elementPadding*2
}