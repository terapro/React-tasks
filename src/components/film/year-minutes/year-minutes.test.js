import React from 'react';
import {YearMinutes} from 'src/components/film/year-minutes/year-minutes.js';
import {shallow} from 'enzyme';

const {year, minutes} = {
  year: '2011-05-05',
  minutes: 111
};

describe('<YearMinutes />', () => {
  describe('Rendering', () => {
    it('should render the component', () => {
      const wrapper = shallow(<YearMinutes year={year} minutes={minutes} />);
      expect(wrapper).toMatchSnapshot();
    });
    it('should render the component with default props', () => {
      const wrapper = shallow(<YearMinutes />);
      expect(wrapper).toMatchSnapshot();
    });
  });

});
