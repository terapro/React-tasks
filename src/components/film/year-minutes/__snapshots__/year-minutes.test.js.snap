// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<YearMinutes /> Rendering should render the component 1`] = `
<div
  className="year-min"
>
  <Label
    content="2011"
    labelBold={true}
    labelUC={false}
    labelWhite={true}
  />
  <Label
    content="111 min"
    labelBold={true}
    labelUC={false}
    labelWhite={true}
  />
</div>
`;
